src/clib/cie_thin_cooling_rate_g_c.c

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************/
+/*
+ * ----------Function to Compute Optically Thin CIE Cooling Rate---------------
+ * 
+ * Written by: Ewan Jones
+ * Date of Creation: 17/2/2021
+ * Date of Last Modification: 17/2/2021
+ * 
+ * Purpose:
+ *  Computes the optically thin cooling rate due to CIE as prescribed in 
+ *  (Ripamonti and Abel 2003).
+ * 
+ * Inputs:
+ *  T : Temperature of the gas (in Kelvin)
+ *  
+ * Outputs:
+ *  cieRate : Optically thin CIE cooling rate.
+ * 
+ * Units:
+ *  erg / s * cm^3 / gram (gram in H2 molecules) --> Units converted to erg/s cm^-3
+ *                                                   by multiplying by rho * rho_mh
+ *                                                   in calc_rates_g.c
+ * 
+ * This code is a refactoring of the original Grackle fortran function of the same name.
+ */
+/*******************************************************************************/
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+
+#include "grackle_macros.h"
+#include "grackle_types.h"
+#include "grackle_chemistry_data.h"
+#include "cie_thin_cooling_rate_tables.h"
+
+//* Define the function which will return the CIE thin cooling rate.
+double cie_thin_cooling_rate_g_c(double T){
```

Comment:
This function isn't actually very long. I recommend moving it to `rate_functions.c` and add the signature to the `grackle_rate_functions.h` header file.